### DISCUSSION TITLE
Read-only access

### DISCUSSION THREAD

**Discussion by fullben** [JanWiemer/jacis#21] [Q&A] [answered]
Am I required to run simple read-only operations as part of a transaction?

Also, what can be considered best practice for implementing read-only operations involving multiple stores?

**Comment by JanWiemer** [comment]
Using the read only API  should also be possible outside of a transaction.
I think I will add this as an example to one of the code snippets in the jacis-codesnippets project...
This way  it should also be possible to access objects from different stores without starting a transaction.
Note that the read only mode supporting objects do not even have to be cloned to access them this way.

May I ask the question how you initially find the JACIS store? :-)

**Reply by fullben** [reply]
I see, so as long as my objects implement `AbstractReadOnlyModeSupportingObject` (and honor the `checkWritable` constraint), read-only access to such data will not result in cloning?

I'm currently working on a project with MicroStream where I'm dealing with a highly mutable data model. I was (and still am I guess) struggling with how to implement a solution with reasonable performance where I neither lose my data mutability nor have to implement transactional support myself... I found this project while checking the MicroStream Hackathon winners for any possible solutions.

**Reply by fullben** [reply]
I'm assuming that this is kind of a big favor to ask, but would you mind publishing a 2.0.7 which includes my accepted PR and the removal of the logging config file? That would really help me out :)

**Comment by JanWiemer** [comment, marked as answer]
* Yes, the reas only mode supporting objects are not cloned when you access them read-only!
* I wish you all success for your project!! At least the intention of the JACIS store was to support transactions for transient data with the best possible performance. Nice that you find the project this way!
* Fortunately I am quite independent with my release policy :-). Therefore I will try to release 2.0.7 as soon as possible (at least within the following days)...
